components/learn/syllabus-tree/syllabus-tree-container.js

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import styled from 'react-emotion';
+
+import TreeView from './syllabus-tree-component';
+
+export default class SyllabusTree extends React.Component {
+  state = {
+    nodeStateTracker: this.props.data.map(() => true),
+  };
+
+  handleClick = i => {
+    this.setState({
+      nodeStateTracker: [
+        ...this.state.nodeStateTracker.slice(0, i),
+        !this.state.nodeStateTracker[i],
+        ...this.state.nodeStateTracker.slice(i + 1),
+      ],
+    });
+  };
+
+  render() {
+    const Container = styled.div`
+      & .node {
+        transition: all 0.5s;
+        border-radius: 3px;
+      }
+
+      & .node:hover,
+      & .info:hover {
+        background-color: rgb(220, 245, 243);
+        cursor: pointer;
+      }
+
+      & .info,
+      & .node {
+        padding: 2px 10px 2px 5px;
+        font: 14px Helvetica, Arial, sans-serif;
+        user-select: none;
+      }
+
+      & .tree-view_arrow {
+        transition: all 0.1s;
+      }
+
+      & .tree-view_arrow-empty {
+        color: yellow;
+      }
+    `;
+
+    return (
+      <Container>
+        {this.props.data.map((node, i) => {
+          const UnitTitle = (
+            <span className="node" key={node.unit.num} onClick={() => this.handleClick(i)}>
+              {node.unit.name}
+            </span>
+          );
+          return (
+            <TreeView
+              key={i}
```

Comment:
Can we assign `key={node.unit.name}` as in line no 54

Reply:
@abiduzz420 i am pushing the latest changes. I will go through your review in some time Thank you for reviewing.

Reply:
Also do you have sometime to work on this pr?

Reply:
@M-ZubairAhmed No, I may not be able to give time due to exams.